### PR TITLE
Adds LineVis for complex data (phase/amplitude)

### PR DESCRIPTION
**Pull request by loichuder**
Work on #310 

image

I kept `ComplexVis` as name for the 2D Complex vis but we can rename it (in `ComplexMap` for example).

### CODE REVIEW

File: src/h5web/toolbar/ComplexLineToolbar.tsx [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+        ariaLabel="Curve type"
+        value={curveType}
+        onChange={(val) => {
+          setCurveType(val as CurveType);
+        }}
+      >
+        <ToggleGroup.Btn label="Line" value={CurveType.LineOnly} />
+        <ToggleGroup.Btn label="Points" value={CurveType.GlyphsOnly} />
+        <ToggleGroup.Btn label="Both" value={CurveType.LineAndGlyphs} />
+      </ToggleGroup>
```
Comment (loichuder) [inline src/h5web/toolbar/ComplexLineToolbar.tsx:98]: Copied from `LineToolbar`

File: src/h5web/vis-packs/core/complex/MappedComplexLineVis.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import { useComplexConfig } from './config';
+import MappedLineVis from '../line/MappedLineVis';
+import { ComplexVisType } from './models';
+
+interface Props {
+  value: H5WebComplex[];
+  dims: number[];
+  dimMapping: DimensionMapping;
+  title: string;
+}
```
Comment (loichuder) [inline src/h5web/vis-packs/core/complex/MappedComplexLineVis.tsx:14]: For now, I declare only the props that are used in `ComplexLineContainer`, I will add the remaining props once I add NeXus support.

File: src/h5web/vis-packs/core/complex/lineConfig.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+import { persist } from 'zustand/middleware';
+import { useState } from 'react';
+import type { ConfigProviderProps } from '../../models';
+import createContext from 'zustand/context';
+import { ComplexLineVisType, ComplexVisType } from './models';
+
+interface ComplexLineConfig {
+  visType: ComplexLineVisType;
+  setVisType: (visType: ComplexLineVisType) => void;
+}
```
Comment (loichuder) [inline src/h5web/vis-packs/core/complex/lineConfig.tsx:11]: As the `visType` is a subset of `ComplexVisType`, I prefered to use a dedicated config for the ComplexLine.

File: src/h5web/vis-packs/core/visualizations.ts [hunk 4]
```diff
@@ -91,7 +95,7 @@ export const CORE_VIS: Record<Vis, CoreVisDef> = {
   },
 
   [Vis.Complex]: {
-    name: Vis.Complex,
+    name: Vis.Heatmap,
```
Comment (loichuder) [inline src/h5web/vis-packs/core/visualizations.ts:98]: Changed `Complex` vis name to `Heatmap`

File: src/h5web/vis-packs/core/visualizations.ts [hunk 5]
```diff
@@ -104,4 +108,15 @@ export const CORE_VIS: Record<Vis, CoreVisDef> = {
...
+  [Vis.ComplexLine]: {
+    name: Vis.Line,
+    Icon: FiActivity,
+    Toolbar: ComplexLineToolbar,
+    Container: ComplexLineVisContainer,
+    ConfigProvider: ComplexLineConfigProvider,
+    supportsDataset: (dataset) => {
+      return hasComplexType(dataset) && hasArrayShape(dataset);
+    },
+  },
```
Comment (axelboc) [inline src/h5web/vis-packs/core/visualizations.ts:121]: This needs to come before `Vis.Complex` so the tab shows up in the right order

File: src/h5web/vis-packs/core/visualizations.ts [hunk 6]
```diff
@@ -32,6 +35,7 @@ export enum Vis {
   Line = 'Line',
   Heatmap = 'Heatmap',
   Complex = 'Complex',
+  ComplexLine = 'Complex Line',
```
Comment (axelboc) [inline src/h5web/vis-packs/core/visualizations.ts:38]: I would remove the space here, to convey that the string is not used as a label.

**Review by axelboc — APPROVED** [review]
Looks good to me!

**Pull request merged**